src/services/schema.tsx

```diff
@@ -134,26 +141,25 @@ const validateValue = (propModel, value, { ignoreMandatory = false }) => {
       };
 
       if (Array.isArray(value) && value.length) {
-        // values could have different class models
         value.forEach((val) => {
-          embeddedModel = schemaDefn.get(val);
+          embeddedModel = schemaDefn.get(val['@class']);
 
           if (embeddedModel) {
-            Object.values(embeddedModel.properties).forEach((subPropModel) => {
+            Object.values(schemaDefn.getProperties(embeddedModel.name)).forEach((subPropModel) => {
               subErrors = valErrorCheck(subPropModel, val, subErrors);
             });
           }
         });
       } else {
         try {
-          embeddedModel = schemaDefn.get(value);
+          embeddedModel = schemaDefn.get(value['@class']);
```

Comment:
non-blocking: this change is not actually necessary as get accepts an object still https://github.com/bcgsc/pori_graphkb_schema/blob/v4.0.0/src/schema.ts#L50-L54